examples/librilight/matching.py

```diff
@@ -0,0 +1,906 @@
+    # Read https://docs.python.org/3/library/multiprocessing.html#multiprocessing.pool.ThreadPool
+    # for more details
+    arguments = []
+    for i in range(len(transcripts)):
+        arguments.append(
+            (i, sourced_text, transcripts_cut_index[i], candidate_matches[i])
+        )
+
+    # Note: Only the levenshtein part runs in the ThreadPool.
+    # other parts are implemented in python, we can not speed them up due to
+    # the GIL
+    num_processes = min(len(arguments), os.cpu_count())
+    with Pool(num_processes) as pool:
+        logging.info("Matching with levenshtein.")
+        async_results = pool.starmap_async(align_worker, arguments)
+        results = async_results.get()
+
+        cut_segment_index: Dict[str, int] = {}
+        cut_list = []
+        for item in results:
+            cut_indexes = item[0]
+            segments = item[1]
+
+            current_cut = batch_cuts[cut_indexes[0]]
+            if current_cut.id not in cut_segment_index:
+                cut_segment_index[current_cut.id] = 0
+
+            for seg in segments:
+                id = f"{current_cut.id}_{cut_segment_index[current_cut.id]}"
+                cut_segment_index[current_cut.id] += 1
+                supervision = SupervisionSegment(
+                    id=id,
+                    channel=current_cut.supervisions[cut_indexes[1]].channel,
+                    language=current_cut.supervisions[cut_indexes[1]].language,
+                    speaker=current_cut.supervisions[cut_indexes[1]].speaker,
+                    recording_id=current_cut.recording.id,
+                    start=seg["start_time"],
+                    duration=seg["duration"],
+                    custom={
+                        "texts": [seg["ref"], seg["hyp"]],
+                        "pre_texts": [seg["pre_ref"], seg["pre_hyp"]],
+                        "begin_byte": seg["begin_byte"],
+                        "end_byte": seg["end_byte"],
+                    },
+                )
+                cut = MonoCut(
+                    id,
+                    start=seg["start_time"],
+                    duration=seg["duration"],
+                    channel=current_cut.channel,
+                    supervisions=[supervision],
+                    recording=current_cut.recording,
+                    custom={"text_path": str(params.book_dir / current_cut.text_path)},
+                )
+                cut_list.append(cut)
+
+        logging.info(f"Writing results.")
+        for cut in cut_list:
+            cuts_writer.write(cut, flush=True)
+        logging.info(f"Write results done.")
+
+
+def main():
+    args = get_args()
+    args.book_dir = Path(args.book_dir)
+    params = get_params()
+    params.update(vars(args))
+
+    raw_cuts = load_manifest_lazy(params.manifest_in)
+    cuts_writer = CutSet.open_writer(params.manifest_out, overwrite=True)
+    batch_cuts = []
+    logging.info(f"Start processing...")
+    for i, cut in enumerate(raw_cuts):
```

Comment:
I suggest that you put the process pool here.
You can make each process write its own file and combine them at the end.